lib/cloud.js

```diff
@@ -183,58 +179,36 @@ class Cloud extends api.events.EventEmitter {
     const isTarget = this.role === 'client';
     if (!isTarget) {
       emitRemoteEvent('state', delta);
-    } else if (impress.nodeId !== delta.from) {
+    } else {
       const application = impress.applications[delta.app];
       if (application && application.state) {
         application.state.emit('change', delta.data, isTarget);
       }
     }
   }
-
 }
 
-class CloudEmitter {
+class CloudEmitter extends api.common.EnhancedEmitter {
 
   constructor(application) {
+    super();
     this.application = application;
 
-    application.frontend.on('*', (eventName, data, isTarget) => {
-      if (!isTarget) {
-        api.cloud.sendEvent(application.name, eventName, data, 'frontend');
-      }
-    });
-
-    application.backend.on('*', (eventName, data, isTarget) => {
-      if (!isTarget) {
-        api.cloud.sendEvent(application.name, eventName, data, 'backend');
-      }
-    });
-
     application.state.on('change', (data, isTarget) => {
       if (!isTarget) {
         api.cloud.sendState(application.name, data);
       }
     });
   }
 
-  sendEventToNode(eventName, data, nodeId) {
```

Comment:
Removal of this method and the ones below breaks backward compatibility and has to be introduced in the next major version. Thus we should probably mark the PR as such, to avoid accidental publish of these changes without the major version increment.

Reply:
Then I think we should also remove `frontend` and `backend` emitters completely?
/cc @tshemsedinov